crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2039,6 +2025,16 @@ impl<'src> Parser<'src> {
     fn parse_type_param(&mut self) -> ast::TypeParam {
         let start = self.node_start();
 
+        // TODO(dhruvmanila): CPython throws an error if `TypeVarTuple` or `ParamSpec`
+        // has bounds:
+        //
+        //    type X[*T: int] = int
+        //             ^^^^^
+        // SyntaxError: cannot use bound with TypeVarTuple
+        //
+        // We should do the same but currently we can't without throwing away the parsed
+        // expression because the AST can't contain it.
+
         if self.eat(TokenKind::Star) {
```

Comment:
Same here. Could we add some more invalid and valid test cases (or did you already do so in another PR)?

For example, the example you mention above would be good (I suspect that fails with a syntax error today)